### ISSUE TITLE
Daily forecast is one day off with AEMET OpenData weather integration

### ISSUE THREAD

**Issue by enriquerollon** [Makin-Things/platinum-weather-card#57]
Thanks for your awesome work! I really like the card; however, I am having an issue with daily forecast and AEMET OpenData weather integration (one of most popular weather services in Spain):

**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser
- [x] I tried another weather card to see if the problem is related to the weather platform

**Release with the issue:** 0.0.4 and 0.0.5b7

**Last working release (if known):** none

**Browser and Operating System:** IOS Home Assistant app, Chrome for Windows

**Description of problem:** There is a one day offset between the data from AEMET OpenData and forecast shown in Platinum Weather Card. This is the comparison between HA Weather Card and Platinum:

<img width="425" alt="image" src="https://user-images.githubusercontent.com/85040979/195709033-39d66562-cc8e-4f5b-a246-eb9308d3c2a9.png">

**Card yaml code:**
```
card_config_version: 5
entity_forecast_high_temp_1: weather.aemet_daily
entity_forecast_icon_1: weather.aemet_daily
entity_forecast_low_temp_1: weather.aemet_daily
entity_pop_1: weather.aemet_daily
section_order:
  - overview
  - extended
  - slots
  - daily_forecast
show_section_extended: false
show_section_overview: false
show_section_slots: false
type: custom:platinum-weather-card
```

**weather.aemet_daily data:**
```
temperature: 18
temperature_unit: °C
humidity: 45
pressure_unit: hPa
wind_bearing: 0
wind_speed: 2
wind_speed_unit: km/h
visibility_unit: km
precipitation_unit: mm
forecast: 
- condition: partlycloudy
  precipitation_probability: null
  datetime: '2022-10-13T22:00:00+00:00'
  wind_bearing: 225
  temperature: 26
  templow: 14
  wind_speed: 5
- condition: sunny
  precipitation_probability: null
  datetime: '2022-10-14T22:00:00+00:00'
  wind_bearing: 180
  temperature: 26
  templow: 14
  wind_speed: 10
- condition: partlycloudy
  precipitation_probability: 25
  datetime: '2022-10-15T22:00:00+00:00'
  wind_bearing: 180
  temperature: 26
  templow: 13
  wind_speed: 10
- condition: rainy
  precipitation_probability: 80
  datetime: '2022-10-16T22:00:00+00:00'
  wind_bearing: 180
  temperature: 24
  templow: 15
  wind_speed: 10
- condition: partlycloudy
  precipitation_probability: 15
  datetime: '2022-10-17T22:00:00+00:00'
  wind_bearing: 135
  temperature: 31
  templow: 14
  wind_speed: 10
- condition: cloudy
  precipitation_probability: 30
  datetime: '2022-10-18T22:00:00+00:00'
  wind_bearing: 225
  temperature: 21
  templow: 13
  wind_speed: 20

attribution: Powered by AEMET OpenData
friendly_name: AEMET daily
```

Screenshots and data are taken at 22:00 aprox GTM+1 (Madrid). 13 October 2022.

**Additional information:** It didn't resolved by #45 

Maybe it’s related with #56. Due to this issue, I noticed that between 00:00 and 00:59 forecast is showed well.

**Comment by Makin-Things** [comment]
This is caused by the timestamps in the forecasts being in UTC time (they should really use the local offset time).
I have attempted to fix it in 0.0.5b8, but it is hard for me to test thoroughly.  Hopefully it doesn't break for others now.

Also I noticed that the AEMET integration is returning null for precipitation instead of 0 for some days(shows as NaN% in the card. This is a bug in that integration that its auther needs to fix.
